src/backup.sh

```diff
@@ -0,0 +1,288 @@
+# This file deals with backing up and restoring kw data files stored under the
+# the KW_DATA_DIR directory. These files are configs, pomodoro reports and
+# kw usage statistics.
+
+include "$KW_LIB_DIR/kwlib.sh"
+include "$KW_LIB_DIR/kw_string.sh"
+include "$KW_LIB_DIR/kwio.sh"
+include "$KW_LIB_DIR/kw_time_and_date.sh"
+
+declare -gA options_values
+decompress_path="$KW_CACHE_DIR/tmp-kw-backup"
+
+# This is the main function that deals with the operations related to backup. It
+# recieves an argument and call other functions based on that.
+function backup()
+{
+  parse_backup_options "$@"
+  if [[ "$?" -gt 0 ]]; then
+    complain "${options_values['ERROR']}"
+    return 22 # EINVAL
+  fi
+
+  if [[ -n "${options_values['BACKUP_PATH']}" ]]; then
+    create_backup "${options_values['BACKUP_PATH']}"
+    return
+  fi
+
+  if [[ -n "${options_values['RESTORE_PATH']}" ]]; then
+    restore_backup "${options_values['RESTORE_PATH']}"
+    return
+  fi
+}
+
+# This function creates a .tar.gz file containing all the data found in
+# KW_DATA_DIR
+#
+# @path Path to the directory in which the compressed file should be stored.
+function create_backup()
+{
+  local path="$1"
+  local flag="$2"
+  local file_name
+  local current_time
+  local ret
+
+  flag=${flag:-'SILENT'}
+
+  if [[ ! -d "$path" ]]; then
+    complain 'We could not find the path'
+    exit 2 # ENOENT
+  fi
+
+  current_time=$(date +%Y-%m-%d_%H-%M-%S)
+  file_name="kw-backup-$current_time.tar.gz"
+  generate_tarball "$KW_DATA_DIR" "$path/$file_name" 'gzip' "$flag"
```

Comment:
@magalilemes 
Could you update this line to use the new version of `generate_tarbal`?